### PR TITLE
Transaction id validation in `getTransaction`

### PR DESCRIPTION
**Pull request by aniav**
I want to merge this change because `getTransaction` is triggering a 500 error when there is an ID of a wrong type passed as input.

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [x] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Improved GraphQL ID validation messages - #14447 by @patrys
 - Add `voucher` to `checkout` query - #14512 by @zedzior
 - Fix draft order voucher assignment - #14336 by @IKarbowiak
+- `getTransaction` will now return an error when queried with an ID of a wrong type - #16042 by @aniav
```
Comment (Air-t) [inline CHANGELOG.md:78]: shouldn't it be added to `main` only?

**Review by Air-t — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (aniav) [reply CHANGELOG.md:78]: I intentionally added it here because we are introducing the fix in 3.19 and then porting to 3.20.

**Review by tomaszszymanski129 — APPROVED** [review]
Other than changelog (see comment above), LGTM.

File: CHANGELOG.md (hunk 1 above)
Comment (aniav) [reply CHANGELOG.md:78]: ok 👍

**Pull request merged**